torch/fx/graph_module.py

```diff
@@ -132,8 +132,7 @@ def __init__(self, root: Union[torch.nn.Module, Dict[str, Any]], graph: Graph):
         """
         super().__init__()
         if isinstance(root, torch.nn.Module):
-            if hasattr(root, 'training'):
-                self.training = root.training
+            self.__dict__ = root.__dict__
```

Comment:
I don't think we want this behavior in the normal constructor. It's probably better to just modify `__deepcopy__` rather than make this change